app/src/main/java/net/skyscanner/backpack/demo/compose/OverlayStory.kt

```diff
@@ -0,0 +1,150 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.overlay.BpkOverlay
+import net.skyscanner.backpack.compose.overlay.BpkOverlayType
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.demo.R
+import net.skyscanner.backpack.demo.components.OverlayComponent
+import net.skyscanner.backpack.demo.meta.ComposeStory
+
+@Composable
+@OverlayComponent
+@ComposeStory
+fun OverlayStory(modifier: Modifier = Modifier) {
+  Column(modifier = modifier) {
+    OverlaySampleRow {
+      DefaultOverlaySample(overlayType = BpkOverlayType.SolidLow)
+      DefaultOverlaySample(overlayType = BpkOverlayType.SolidMedium)
+      DefaultOverlaySample(overlayType = BpkOverlayType.SolidHigh)
+    }
+    OverlaySampleRow {
+      DefaultOverlaySample(overlayType = BpkOverlayType.TopLow)
+      DefaultOverlaySample(overlayType = BpkOverlayType.TopMedium)
+      DefaultOverlaySample(overlayType = BpkOverlayType.TopHigh)
+    }
+    OverlaySampleRow {
+      DefaultOverlaySample(overlayType = BpkOverlayType.BottomLow)
+      DefaultOverlaySample(overlayType = BpkOverlayType.BottomMedium)
+      DefaultOverlaySample(overlayType = BpkOverlayType.BottomHigh)
+    }
+    OverlaySampleRow {
+      DefaultOverlaySample(overlayType = BpkOverlayType.LeftLow)
+      DefaultOverlaySample(overlayType = BpkOverlayType.LeftMedium)
+      DefaultOverlaySample(overlayType = BpkOverlayType.LeftHigh)
+    }
+    OverlaySampleRow {
+      DefaultOverlaySample(overlayType = BpkOverlayType.RightLow)
+      DefaultOverlaySample(overlayType = BpkOverlayType.RightMedium)
+      DefaultOverlaySample(overlayType = BpkOverlayType.RightHigh)
+    }
+    OverlaySampleRow {
+      DefaultOverlaySample(overlayType = BpkOverlayType.Vignette)
+      WithTextOverlaySample(overlayType = BpkOverlayType.SolidHigh)
+    }
+  }
+}
+
+@Composable
+fun OverlaySampleRow(
+  modifier: Modifier = Modifier,
+  content: @Composable RowScope.() -> Unit,
+) {
+  Row(
+    content = content,
+    horizontalArrangement = Arrangement.SpaceEvenly,
+    modifier = modifier.fillMaxWidth(),
+  )
+}
+
+@Composable
+fun DefaultOverlaySample(
+  overlayType: BpkOverlayType,
+  modifier: Modifier = Modifier,
+) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    BpkOverlay(
+      modifier = modifier
+        .width(110.dp)
```

Comment:
Can we use weight modifier here, and specify the height using aspectRatio modifier?
In order to use weight, your function should be either an extension of RowScope `RowScope.DefaultOverlaySample` or, better, the weight should be added externally and passed to the function